packages/api/src/codegen/languages/typescript/index.ts

```diff
@@ -197,23 +197,15 @@ export default class TSGenerator extends CodeGenerator {
   async compile(storage: Storage, opts: InstallerOptions = {}): Promise<void> {
     const installDir = storage.getIdentifierStorageDir();
 
-    await execa('npx', ['tsup'], {
+    await execa('npm', ['pkg', 'set', 'scripts.prepare=tsup'])
+      .then(res => handleExecSuccess(res, opts))
+      .catch(err => handleExecFailure(err, opts));
+
+    await execa('npm', ['run', 'prepare'], {
       cwd: installDir,
     })
-      .then(res => {
-        if (opts.dryRun) {
-          (opts.logger ? opts.logger : logger)(res.command);
-          (opts.logger ? opts.logger : logger)(res.stdout);
-        }
-      })
-      .catch(err => {
-        if (opts.dryRun) {
-          (opts.logger ? opts.logger : logger)(err.message);
-          return;
-        }
-
-        throw err;
-      });
+      .then(res => handleExecSuccess(res, opts))
+      .catch(err => handleExecFailure(err, opts));
```

Comment:
might be nice to write a small wrapper for `execa()` that handles these success and failure states instead of having to copy-paste this code everywhere but i don't really have the energy at the moment to try to figure out how to clone all of execa's parameter typing into a wrapper